src/sql/workbench/services/objectExplorer/browser/asyncServerTreeRenderer.ts

```diff
@@ -0,0 +1,295 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the Source EULA. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import 'vs/css!./media/objectTypes/objecttypes';
+
+import * as dom from 'vs/base/browser/dom';
+import { localize } from 'vs/nls';
+import { ConnectionProfileGroup } from 'sql/platform/connection/common/connectionProfileGroup';
+import { ConnectionProfile } from 'sql/platform/connection/common/connectionProfile';
+import { IConnectionManagementService } from 'sql/platform/connection/common/connectionManagement';
+import { TreeNode } from 'sql/workbench/services/objectExplorer/common/treeNode';
+import { iconRenderer } from 'sql/workbench/services/objectExplorer/browser/iconRenderer';
+import { URI } from 'vs/base/common/uri';
+import { ITreeRenderer, ITreeNode } from 'vs/base/browser/ui/tree/tree';
+import { FuzzyScore } from 'vs/base/common/filters';
+import { Disposable } from 'vs/base/common/lifecycle';
+import { IInstantiationService } from 'vs/platform/instantiation/common/instantiation';
+import { IKeyboardNavigationLabelProvider } from 'vs/base/browser/ui/list/list';
+import { IListAccessibilityProvider } from 'vs/base/browser/ui/list/listWidget';
+import { ServerTreeRenderer } from 'sql/workbench/services/objectExplorer/browser/serverTreeRenderer';
+import { ServerTreeElement } from 'sql/workbench/services/objectExplorer/browser/asyncServerTree';
+
+class ConnectionProfileGroupTemplate extends Disposable {
+	private _root: HTMLElement;
+	private _nameContainer: HTMLElement;
+
+	constructor(
+		container: HTMLElement
+	) {
+		super();
+		container.parentElement.classList.add('server-group');
+		container.classList.add('server-group');
+		this._root = dom.append(container, dom.$('.server-group'));
+		this._nameContainer = dom.append(this._root, dom.$('span.name'));
+	}
+
+	set(element: ConnectionProfileGroup) {
+		let rowElement = findParentElement(this._root, 'monaco-list-row');
+		if (rowElement) {
+			if (element.color) {
+				rowElement.style.background = element.color;
+			} else {
+				// If the group doesn't contain specific color, assign the default color
+				rowElement.style.background = '#515151';
```

Comment:
You should be able to get this value from the theme service

Reply:
This is for the server groups which have defined colors - themes shouldn't apply to that.

Reply:
Should the default color be defined somewhere then?

Reply:
Yeah not a bad idea - I'll make it so they all reference a single defined value.